util/config/env.go

```diff
@@ -46,8 +48,8 @@ func loadFlags() error {
 	// pkg shellquota doesn't recognize `=` so that the opts in format `foo=bar` could not work.
 	// issue ref: https://github.com/argoproj/argo-cd/issues/6822
 	for k, v := range flags {
-		if strings.Contains(k, "=") && strings.Count(k, "=") == 1 && v == "true" {
-			kv := strings.Split(k, "=")
+		if strings.Contains(k, "=") && v == "true" {
+			kv := strings.SplitN(k, "=", 2)
```

Comment:
Certainly! The change in the code is addressing a specific issue with parsing key-value pairs where the value contains the '=' character.

Originally, the code was checking for keys that contain exactly one '=' character and where the associated value is "true". It then split the key into a key-value pair using `strings.Split`, which splits the string into a slice of strings using the '=' as a delimiter. However, this approach would fail if the value itself contained an '=' character, as it would incorrectly split the value as well.

The modification replaces `strings.Split` with `strings.SplitN`, specifying a maximum number of splits to 2. This means that the function will split the string at the first occurrence of the '=' character, resulting in exactly two substrings: one for the key and one for the value, regardless of whether the value contains additional '=' characters.

This change ensures that even if the value contains the '=' character, such as in the provided test case `--header='Strict-Transport-Security: max-age=31536000'`, it will be correctly parsed into a key-value pair without splitting the value at every '=' character. The key will be `--header` and the value will be `'Strict-Transport-Security: max-age=31536000'`, which is the desired behavior for correctly handling such command-line arguments.

Reply:
Straight out of ChatGPT :joy:

Reply:
😂 You got me! Indeed, I enlisted ChatGPT's help for articulating my thoughts more clearly.